specs/xaml-resource-references-tracing-spec.md

```diff
@@ -0,0 +1,165 @@
+Tracing XAML resource reference lookup failures
+===
+
+# Background
+
+Failure to resolve a XAML resource reference (`{StaticResource}` or `{ThemeResource}`) is one of 
+the most common causes of app crashes. Such failures manifest as a stowed exception with the 
+message "Cannot find a Resource with the Name/Key *foo*", and generally fall into one of two 
+buckets:
+
+1. The referenced resource legitimately does not exist, e.g. the referenced key is misspelled or 
+the intended matching resource has not been added to the app.
+2. The referenced resource *does* exist in the app, but it is not reachable from the reference 
+at run-time.
+
+Unfortunately, the stowed exception message is the *only* information provided about the error, 
+which makes it difficult or, more often, outright impossible to debug. This spec describes a new 
+API on the `DebugSettings` class that developers can use to access more detailed information about 
+the resource reference lookup failure.
+
+# API Pages
+
+_(Each of the following L2 sections correspond to a page that will be on docs.microsoft.com)_
+
+## DebugSettings.XamlResourceReferenceFailed Event
+
+Occurs when a [XAML resource reference](https://learn.microsoft.com/en-us/windows/apps/design/style/xaml-resource-dictionary) cannot be resolved.
+
+_Spec note: is there a version of the 'ResourceDictionary and XAML resource references' article that 
+links to the Windows App SDK documentation for relevant APIs?_
+
+```c#
+public event TypedEventHandler<DebugSettings,XamlResourceReferenceFailedEventArgs> 
+XamlResourceReferenceFailed
+
+```
+
+### Remarks
+
+`IsXamlResourceReferenceTracingEnabled` must be `true` and there must be a debugger attached to 
+the app process in order for `XamlResourceReferenceFailed` to fire and for tracing to appear in 
```

Comment:
If there's no debugger attached, the event isn't even raised? Why not? (Does BindingFailed behave that way?)

Reply:
BindingFailed behaves the same way. But you've got me thinking... designing it this way prevents a developer from logging the event information to their own reporting infrastructure. My original reasoning was that this is only useful as a debugging aid ergo without a debugger attached it's not useful but it's clear now that's faulty reasoning. I think it's worth making this event fire in all circumstances; BindingFailed is potentially noisy because it's non-fatal, but XamlResourceReferenceFailed doesn't have the same concern because the developer needs to go to some effort to make it non-fatal.